### PR TITLE
Fix: allow users with view only acces to use the queries in Query Results

### PR DESCRIPTION
**Pull request by arikfr**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Description

Allow users who have View Only access to a data source using its queries in a Query Results query.

## Related Tickets & Documents

This does not solve the use case described in #4111, but a limited scope of it (where you have View Only access vs. Full Access).

### CODE REVIEW

File: redash/query_runner/query_results.py [hunk 1]
```diff
@@ -34,9 +35,12 @@ def _load_query(user, query_id):
     if user.org_id != query.org_id:
         raise PermissionError("Query id {} not found.".format(query.id))
 
-    if not has_access(query.data_source, user, not_view_only):
-        raise PermissionError(u"You are not allowed to execute queries on {} data source (used for query id {}).".format(
-            query.data_source.name, query.id))
+    # TODO: this duplicates some of the logic we already have in the redash.handlers.query_results.
+    # We should merge it so it's consistent.
+    if not has_access(query.data_source, user, view_only):
```
Comment (arikfr) [inline redash/query_runner/query_results.py]: This is the actual change. The rest is my auto formatter :grimacing:

**Comment by rauchy** [comment]
LGTM. It's a very subtle assumption, so I believe it needs a very descriptive test.

File: tests/query_runner/test_query_results.py [hunk 2]
```diff
@@ -135,6 +211,15 @@ def test_returns_query(self):
         loaded = _load_query(user, query.id)
         self.assertEquals(query, loaded)
 
+    def test_returns_query_when_user_has_view_only_access(self):
```
Comment (arikfr) [inline tests/query_runner/test_query_results.py]: New test.

File: redash/query_runner/query_results.py [hunk 3]
```diff
@@ -34,9 +35,12 @@ def _load_query(user, query_id):
...
+        raise PermissionError(
+            u"You are not allowed to execute queries on {} data source (used for query id {})."
```
Comment (rauchy) [inline redash/query_runner/query_results.py]: Is this error message still accurate after this change?

Comment (arikfr) [reply redash/query_runner/query_results.py]: https://github.com/getredash/redash/pull/4112/commits/32ba5d76790871175934db016a4f34c08d192a4e

Comment (arikfr) [reply redash/query_runner/query_results.py]: Oops: https://github.com/getredash/redash/pull/4112/commits/bb4f76808e206549d5397bc7e14fa852583255a9.

**Pull request merged**